### PR TITLE
Parallel Test Handler

### PR DESCRIPTION
**Pull request by 0xPoly**
NM now can run multiple tests at the same time. An overview of which are running tests at any given time is available on the 'status' page.

### CODE REVIEW

**Comment by 0xPoly** [comment]
oh, closes #25

**Comment by 0xPoly** [comment]
closes #25

File: configure.js [hunk 1]
```diff
@@ -16,17 +17,17 @@ exports.getState = function() {
 
 exports.init = function() {
     var templates = {
-      status: fs.readFileSync(__dirname + '/templates/status.tmpl').toString(),
+      status_: fs.readFileSync(__dirname + '/templates/status.tmpl').toString(),
```
Comment (bassosimone) [inline configure.js:20]: y s/status/status_/ ?

File: exec.js [hunk 2]
```diff
@@ -100,9 +95,10 @@ var collectArgs = function(state) {
     }
 
     // gets rid of any variables that have not been specified
-    command = command.replace(new RegExp(/\$[a-z][a-z0-9_]*\s/g), "");
+    command = executable + " " + command.replace(new RegExp(/\$[a-z][a-z0-9_]*\s/g), "");
 
-    dialog.showMessageBox({"type": "info", "message": command, "buttons": ["close"]});
+    // TODO breaks compatability with windows, use fs.seperator instead
```
Comment (bassosimone) [inline exec.js]: typo: compatability -> compatibility

File: handler.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+/* handler.js
+ * Manages tests running in parallel
```
Comment (bassosimone) [inline handler.js]: Thanks :)

<bikesh>Better not to add the file name because this is not robust to renames</>

File: handler.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+var child_process = require('child_process');
+var configure = require('./configure.js');
+
+exports.startTest = function(name_, command, directory) {
+    var process = child_process.exec(command, { cwd : directory });
+    var container = { name : name_,
+                      test: process };
+
+    process.stdout.on('data', function (data) {
+          console.log('stdout: ' + data);
```
Comment (bassosimone) [inline handler.js]: Here indentation is not consistent.

File: handler.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+}
+
+exports.getFinishedTests = function() {
+}
+
+exports.updateTests = function(state) {
+    state.status_.set("runningTests", tests);
+}
+
+var tests = [];
```
Comment (bassosimone) [inline handler.js]: Can you move this at top, please?

File: handler.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        exports.updateTests(configure.getState());
+    });
+
+    tests.push(container);
+
+    return process;
+}
+
+exports.killTest = function(test) {
+    test.kill('SIGINT');
```
Comment (bassosimone) [inline handler.js:30]: In the future we may want to SIGKILL if the test does not respond for too much time.

(BTW wondering whether SIGINT and SIGKILL are portable in Node)

**Comment by bassosimone** [comment]
Hi @0xPoly, I'm now done with my review! Overall I think this is good stuff that only needs some minor tweaks to be good to go.

File: configure.js (hunk 1 above)
Comment (0xPoly) [inline configure.js:20]: turns out 'status' is a reserved keyword in js

File: exec.js (hunk 2 above)
Comment (0xPoly) [inline exec.js]: I really should add a spell check plugin to vim

File: handler.js (hunk 6 above)
Comment (0xPoly) [inline handler.js:30]: I'll add an issue to test portability sometime in the future

Comment (0xPoly) [inline handler.js:30]: added #33

File: configure.js (hunk 1 above)
Comment (bassosimone) [inline configure.js:20]: Ah, didn't know this!

File: handler.js (hunk 6 above)
Comment (bassosimone) [inline handler.js:30]: Thank you!

**Comment by bassosimone** [comment]
Great, merging now!

**Pull request merged**